src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -2168,12 +2168,16 @@ QString QgsProcessingParameters::parameterAsDatabaseTableName( const QgsProcessi
 
 QgsPointCloudLayer *QgsProcessingParameters::parameterAsPointCloudLayer( const QgsProcessingParameterDefinition *definition, const QVariantMap &parameters, QgsProcessingContext &context )
 {
-  return qobject_cast< QgsPointCloudLayer *>( parameterAsLayer( definition, parameters, context, QgsProcessingUtils::LayerHint::PointCloud ) );
+  QgsProcessing::LayerOptionsFlags optionsFlags;
+  optionsFlags |= QgsProcessing::LayerOptionsFlag::SkipIndexGeneration;
```

Comment:
I don't think we should always set this flag here, rather we should add a flag argument to QgsProcessingParameters::parameterAsPointCloudLayer and make the particular algorithm decide on whether or not to do the indexing. I'm thinking some future algorithms may NEED the index to be usable....